From 68ae4f10232d731304de7a707501a96dc25d9c22 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 08 Apr 2022 09:20:38 +0300
Subject: [PATCH] [SW-79695] gaudi3: update fw version location

As per the updated efuse bits definition for gaudi3, we can
remove alternate locations (spi, spi otp) for fw version.

Fw version will always be read from efuse.

Change-Id: I01d1cec3710ffb668be4a3aa554fecb09e3205c9
---

diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index ac4a1c1..b785bf9 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -17,10 +17,6 @@
 #define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_EROM	0x1
 
 #define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(11)
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION		7
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK		0x3
-
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI		BIT(8)
 
 #define EFUSE_SYSCFG_EFUSE_LOCATION_MASK		BIT(25)
 #define EFUSE_SYSCFG_SKU_MASK				BIT(26)
@@ -28,6 +24,8 @@
 /* Following bitmaps are used in common code, hence maintained for compilation */
 #define EFUSE_SYSCFG_REVOKATION_LOCATION		0
 #define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK		0x0
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION		0
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK		0x0
 
 /*
  * As per recommendation from Security Architect, if FUSE is not available,
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index b6f5cdf..49df10d 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -192,26 +192,12 @@
 
 void security_load_fw_version(void)
 {
-	uint8_t fw_version_sec = 0;
-	enum security_location fw_version_location;
-
-	fw_version_location = ((arcboot_rom_global.config_fuse >>
-			       EFUSE_SYSCFG_FW_VERSION_LOCATION) &
-			       EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK);
+	uint8_t fw_version_sec = efuse_otp_get_fw_version_security();
 
 	arcboot_rom_status_update(PROGRESS_VER_LOC);
+	/* REVOKE BITS LOCATION: EFUSE */
 	arcboot_rom_status_update_location(PROGRESS_VER_LOC_0,
-					   fw_version_location);
-
-	switch (fw_version_location) {
-	/* TODO: Add other locations if required for gaudi3 */
-	/* TODO: Confirm enumeration values for gaudi3 */
-	case ASIC_EFUSE:
-		fw_version_sec = efuse_otp_get_fw_version_security();
-		break;
-	default:
-		return;
-	}
+					   ASIC_EFUSE);
 
 	/* store minimum allowed security version in PSOC */
 	regmap_writel(SECURITY_FW_VERSION_FROM_SECURITY_BYTE(fw_version_sec),
